hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/OFSPath.java

```diff
@@ -238,4 +248,23 @@ static String getTempMountBucketNameOfCurrentUser() throws IOException {
     String username = UserGroupInformation.getCurrentUser().getUserName();
     return getTempMountBucketName(username);
   }
+
+  /**
+   * Return trash root for the given path.
+   * @return trash root for the given path
+   */
+  public Path getTrashRoot() {
+    try {
+      String username = UserGroupInformation.getCurrentUser().getUserName();
+      URI uri = new URIBuilder().setScheme(OZONE_OFS_URI_SCHEME)
+          .setHost(authority).setPath(OZONE_URI_DELIMITER + volumeName +
```

Comment:
NIT: we can use Path API to create the path safely instead of string concat.

Reply:
Do you mean using constructor `new Path(prevPath, "child")` and building the path step by step?

Reply:
Attempted in 53c19f1f1f543978f7e381d2160e35415756a859.

I also explored [`Paths.get()`](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Paths.html) but it returns `java.nio.file.Paths`.